docs/en/guides/developer/working-with-json/json-load-data.md

```diff
@@ -0,0 +1,129 @@
+---
+slug: /en/guides/developer/working-with-json/json-load-data
+sidebar_label: Loading JSON
+sidebar_position: 1
+description: Loading JSON into ClickHouse
+---
+
+# Loading JSON in 5 steps
+
+This guide walks through the process to load logging data that is 
+in a JSON formatted file in S3.  In order to do this:
+- Examine the file format by selecting one row using the S3 function
+- Create a table to store the data in ClickHouse
+- Load a single row of nested JSON
+- Verify the correct storage of the nested JSON
+- Import the dataset from S3
+
+## Examine the structure of the JSON file
+Examine one record from the log file in S3.  The `s3` function
+retrieves and decompresses the file and allows querying the file
+in S3 without loading it.
+```sql
+SELECT * FROM s3('https://datasets-documentation.s3.eu-west-3.amazonaws.com/http/documents-01.ndjson.gz', 
+'JSONEachRow') LIMIT 1;
+```
+```response
+┌─@timestamp─┬─clientip───┬─request──────────────────────────────────────────────────────────┬─status─┬──size─┐
+│  893964617 │ 40.135.0.0 │ {'method':'GET','path':'/images/hm_bg.jpg','version':'HTTP/1.0'} │    200 │ 24736 │
+└────────────┴────────────┴──────────────────────────────────────────────────────────────────┴────────┴───────┘
+```
+Note that the `response` field contains nested JSON, it is more
+efficient for the users of the log data if that JSON is also extracted
+into separate fields. The next two steps will be performed with this
+optimization in mind.
+
+## Create a ClickHouse table
+To maximize the usefulness of the data we
+need to extract the nested `method`, `path`, and `version` fields under `request`.  To prepare for this, create a table including those nested fields:
+```sql
+CREATE table http
+(
+   `@timestamp` Int32 EPHEMERAL 0,
+   clientip     IPv4,
+# highlight-next-line
+   request Nested(method LowCardinality(String), path String, version LowCardinality(String)),
+   status       UInt16,
+   size         UInt32,
+   timestamp    DateTime DEFAULT toDateTime(`@timestamp`)
+) ENGINE = MergeTree() ORDER BY (status, timestamp);
+```
+
+## Insert one row
+```sql
+INSERT INTO http (`@timestamp`, clientip, request.method, request.path, request.version, status, size) SELECT
+    `@timestamp`,
+    clientip,
+    [request['method']],
+    [request['path']],
+    [request['version']],
+    status,
+    size
+FROM s3('https://datasets-documentation.s3.eu-west-3.amazonaws.com/http/documents-01.ndjson.gz', 'JSONEachRow')
+LIMIT 1
```

Comment:
That's not a good way to use type `Nested`. `Nested` should be used when we have object with nested **arrays** of data. For example:
`{"id" : 1, "nested" : {"field1" : [1, 2], "field2" : ["value1", "value2"]}}`.
It doesn't make sense to use when we have just simple nested object with string values as we have here.
If we are sure that `request` field always have only these three nested fields `method` and `path`, `version` and they are always String, we can:
1) Just leave it as a Map type:
https://pastila.nl/?0007b917/57c1b3d174e4b1f9247777d433eacbd9
2) Create a separate field for each nested fields `method` and `path`, `version`:

 use just a `Tuple(method LowCardinality(String), path String, version LowCardinality(String))`. Like this: